akvo/rsr/views/project.py

```diff
@@ -152,6 +152,18 @@ def directory(request):
 # Project main
 ###############################################################################
 
+def _check_project_viewing_permissions(user, project):
+    """
+    Checks if the user can view a project, otherwise raises a PermissionDenied exception.
+
+    A user can view any public project, but when a project is private or not published the user
+    should be logged in and able to make changes to the project (e.g. be an admin of the project).
+    """
+    if (project.is_private or not project.is_published()) and \
+            user.is_anonymous() or \
+            (not user.is_anonymous() and not user.has_perm('rsr.change_project', project)):
+        raise PermissionDenied
+
```

Comment:
I'm guessing this could be shortened thus:

```python
    if ((project.is_private or not project.is_published()) 
            and not user.has_perm('rsr.change_project', project)):
        raise PermissionDenied
```
since anonymous users never have rsr.change_project perm.